calibr8/core.py

```diff
@@ -525,16 +531,17 @@ def loglikelihood(
                 if not name:
                     raise ValueError("A `name` must be specified for the PyMC likelihood.")
                 rv = self.pymc_dist(name, **self.to_pymc(*params), observed=y, **dist_kwargs or {})
+                return pmodel.logp([rv], sum=True)
             else:
                 rv = self.pymc_dist.dist(**self.to_pymc(*params), **dist_kwargs or {})
-            # The API to get log-likelihood tensors differs between PyMC versions
-            if pm.__version__[0] == "3":
-                if isinstance(rv, pm.model.ObservedRV):
-                    return rv.logpt.sum()
-                elif isinstance(rv, pm.Distribution):
-                    return rv.logp(y).sum()
-            else:
-                return pm.joint_logpt(rv, y, sum=True)
+                # The out-of-modelcontext API for log-likelihoods differs between PyMC v4 and v5
+                y_tensor = pt.as_tensor(y, dtype=rv.type.dtype)
+                if pm.__version__[0] == "4":
+                    return pm.joint_logp(rv, y, sum=True)
+                elif int(pm.__version__[0]) >= 5:
+                    return pm.logprob.joint_logprob({rv: y_tensor}, sum=True)
```

Comment:
So in PyMC v5 this is only working with tensor inputs?

Reply:
Yes, the previous `pm.joint_logp` was a wrapper that apparently did this automatically..